cmd/bosun/expr/funcs.go

```diff
@@ -440,6 +447,86 @@ func Filter(e *State, T miniprofiler.Timer, series *Results, number *Results) (*
 	return series, nil
 }
 
+func quickSelect(data sort.Interface, k int, max bool) {
```

Comment:
My vote would be to just use sort.Sort() like you mentioned in the comments to keep it simple